kernel/src/transaction.rs

```diff
@@ -57,26 +55,22 @@ pub fn add_files_schema() -> &'static SchemaRef {
 /// ```
 pub struct Transaction {
     read_snapshot: Arc<Snapshot>,
-    operation: Option<String>,
-    commit_info: Option<Arc<dyn EngineData>>,
+    commit_info: Box<CommitInfo>,
     add_files_metadata: Vec<Box<dyn EngineData>>,
     // NB: hashmap would require either duplicating the appid or splitting SetTransaction
     // key/payload. HashSet requires Borrow<&str> with matching Eq, Ord, and Hash. Plus,
     // HashSet::insert drops the to-be-inserted value without returning the existing one, which
     // would make error messaging unnecessarily difficult. Thus, we keep Vec here and deduplicate in
     // the commit method.
     set_transactions: Vec<SetTransaction>,
-    // commit-wide timestamp (in milliseconds since epoch) - used in ICT, `txn` action, etc. to
-    // keep all timestamps within the same commit consistent.
-    commit_timestamp: i64,
```

Comment:
unclear whether or not we want to remove these - at first glance, i think no - we can track these as top-level items then at the end bundle them all up in commit info? since we need to e.g. update operation or reference timestamp